Assets/Tests/PlayModeTests/SessionTests.cs

```diff
@@ -0,0 +1,301 @@
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                ServerUrl = _serverUrl,
+                AppKey = _appKey,
+            };
+
+            configuration.DisableAutomaticSessionTracking();
+            Countly.Instance.Init(configuration);
+            Assert.IsNotNull(Countly.Instance.Session);
+            Assert.AreEqual(0, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+        }
+
+        /// <summary>
+        /// It validates the functionality of 'BeginSessionAsync' when session consent is given after init and automatic session tracking is disable.
+        /// </summary>
+        [Test]
+        public void TestSessionBegin_ConsentGivenAfterInitWithDisableTracking()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                ServerUrl = _serverUrl,
+                AppKey = _appKey,
+                RequiresConsent = true
+            };
+
+            configuration.DisableAutomaticSessionTracking();
+            Countly.Instance.Init(configuration);
+            Assert.IsNotNull(Countly.Instance.Session);
+            Assert.AreEqual(1, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+            Countly.Instance.Session._requestCountlyHelper._requestRepo.Clear();
+
+            Countly.Instance.Consents.GiveConsent(new Consents[] { Consents.Sessions });
+
+            Assert.AreEqual(0, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+           
+        }
+
+        /// <summary>
+        /// It validates the functionality of 'BeginSessionAsync' when session consent is given after init and session had begun before.
+        /// </summary>
+        [Test]
+        public void TestSessionBegin_ConsentGivenAfterSessionBegin()
+        {
+            CountlyConfiguration configuration = new CountlyConfiguration {
+                ServerUrl = _serverUrl,
+                AppKey = _appKey,
+                RequiresConsent = true
+            };
+
+            Countly.Instance.Init(configuration);
+            Assert.IsNotNull(Countly.Instance.Session);
+            Assert.AreEqual(1, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+            Countly.Instance.Session._requestCountlyHelper._requestRepo.Clear();
+
+            Countly.Instance.Consents.GiveConsent(new Consents[] { Consents.Sessions });
+            Assert.AreEqual(1, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+            Countly.Instance.Session._requestCountlyHelper._requestRepo.Clear();
+
+            Countly.Instance.Consents.RemoveConsent(new Consents[] { Consents.Sessions });
+            Assert.AreEqual(0, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
+
+            Countly.Instance.Consents.GiveConsent(new Consents[] { Consents.Sessions });
+            Assert.AreEqual(0, Countly.Instance.Session._requestCountlyHelper._requestRepo.Count);
```

Comment:
why is this "0"? a session should be started

Reply:
On remving session consent we don't end session.
If session has be strated before, after giving back 'session consent'  session begin request shouldn't send.

Reply:
after merging the consent request ticket, it will become 1 as on changing request, consent request will add into RQ.

Reply:
Doesn't matter that a "end session" request wasn't sent. If automatic sessions are enabled, "begin session" request should be sent after giving consent